src/sentry/static/sentry/app/components/dropdownMenu.jsx

```diff
@@ -0,0 +1,227 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+class DropdownMenu extends React.Component {
+  static propTypes = {
+    onOpen: PropTypes.func,
+    onClose: PropTypes.func,
+    onClickOutside: PropTypes.func,
```

Comment:
What's the use case for exposing `onClickOutside`? Is this gonna be used for something in the future?

Reply:
I'll add some comments for it... it's if DropdownMenu is controlled we'll want some callbacks for its parent component.